### PR TITLE
{tools}[intel/2017b] XCfun v20180122

### PR DESCRIPTION
**Pull request by smoors**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+description = """ XCFun is a library of DFT exchange-correlation (XC) functionals.
+ It is based on automatic differentiation and can therefore generate arbitrary order
+ derivatives of these functionals. """
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+
+toolchainopts = {'pic': True}
+
+source_urls = ['https://github.com/dftlibs/xcfun/archive/']
+sources = ['%s.tar.gz' % commit]
```
Comment (boegel) [inline easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb]: @smoors Please let EasyBuild rename the tarball on download (cfr. http://easybuild.readthedocs.io/en/latest/Writing_easyconfig_files.html#common-easyconfig-param-sources-alt)

```python
sources = [{'download_filename': '%s.tar.gz' % commit, 'filename': SOURCE_TAR_GZ}]
```

File: easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+easyblock = 'MakeCp'
+
+name = 'XCfun'
+version = '20180122'
+commit = 'acd11bfbdcc2d2fb9c100111545bd479e6e2c452'
```
Comment (boegel) [inline easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb]: @smoors Usually we stick to the 7-digit short commit format:

```python
commit = 'acd11bf'
```

File: easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'http://dftlibs.org/xcfun/'
+description = """ XCFun is a library of DFT exchange-correlation (XC) functionals.
+ It is based on automatic differentiation and can therefore generate arbitrary order
+ derivatives of these functionals. """
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+
```
Comment (boegel) [inline easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb]: @smoors style nitpicking: please drop empty line

File: easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+toolchainopts = {'pic': True}
+
+source_urls = ['https://github.com/dftlibs/xcfun/archive/']
+sources = ['%s.tar.gz' % commit]
+
```
Comment (boegel) [inline easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb]: drop empty line above/below `patches = ` line

File: easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+patches = ['%(name)s-%(version)s-shared_library-intel.patch']
+
+checksums = [
+    # acd11bfbdcc2d2fb9c100111545bd479e6e2c452.tar.gz
+    '1dd1bac10fd12682ac0b65c51fd1fd5928104d1210165ef7d9e433ec99090e4e',
+    # XCfun-20180122-shared_library-intel.patch
+    '4386590cffcae74e0ee443d2cbff8e927a2de026a5e0842b7d268a9f4f490054',
+]
+
+
```
Comment (boegel) [inline easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb]: @smoors please avoid duplicate empty lines

File: easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+dependencies = [
+    ('Python', '2.7.14'),
+]
+
+prebuildopts = 'cp Makefile.intel Makefile && '
+
+files_to_copy = ['lib', 'include', 'doc', 'test']
+
+sanity_check_paths = {
+    'files': ['lib/libxcfun.%s' % x for x in ['a', SHLIB_EXT]] + ['include/xcfun.h'],
```
Comment (boegel) [inline easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb]: using a list comprehension here is a bit overkill, better 'unroll' it (and sort alphabetically):

```python
'files': ['include/xcfun.h', 'lib/libxcfun.a', 'lib/libxcfun.%s' % SHLIB_EXT],

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2627.swalot.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/b788d0160406116497f825a106ae309f for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2050.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/54ae8ce0b6d11846359be28bec8c0428 for a full test report.

File: easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+ derivatives of these functionals. """
+
+toolchain = {'name': 'intel', 'version': '2017b'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://github.com/dftlibs/xcfun/archive/']
+sources = [{'download_filename': '%s.tar.gz' % commit, 'filename': SOURCE_TAR_GZ}]
+patches = ['%(name)s-%(version)s-shared_library-intel.patch']
+checksums = [
+    # acd11bfbdcc2d2fb9c100111545bd479e6e2c452.tar.gz
```
Comment (boegel) [inline easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb]: @smoors Please also update this comment, should be renamed source tarball (`XCfun-20180122.tar.gz`)

File: easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb [hunk 8]
```diff
@@ -19,7 +19,7 @@ patches = ['%(name)s-%(version)s-shared_library-intel.patch']
 checksums = [
     # acd11bfbdcc2d2fb9c100111545bd479e6e2c452.tar.gz
     '1dd1bac10fd12682ac0b65c51fd1fd5928104d1210165ef7d9e433ec99090e4e',
-    # XCfun-20180122-shared_library-intel.patch
+    # XCfun-20180122.tar.gz
```
Comment (boegel) [inline easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb]: @smoors You fixed the wrong comment... ;)

Comment (smoors) [reply easybuild/easyconfigs/x/XCfun/XCfun-20180122-intel-2017b-Python-2.7.14.eb]: oops haast en spoed en zo :)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2672.swalot.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/3cc4ec5954aaf24e2e3886cf369b8242 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2025.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/4520b2913612c52f455fc3d2fc9b97a0 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @smoors!

**Pull request merged**